src/vdc_collection.rs

```diff
@@ -0,0 +1,130 @@
+use super::storage_manager::*;
+use serde_derive::{Deserialize, Serialize};
+
+/// Supported credential formats.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialFormat {
+    MsoMdoc,
+    JwtVcJson,
+    LdpVc,
+    Other(String), // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String), // For ease of expansion.
+}
+
+/// An individual credential.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Credential {
+    id: String, // Probably a UUID, perhaps we should make it one?
+    format: CredentialFormat,
+    ctype: CredentialType,
+    payload: Vec<u8>, // The actual credential.
+}
+
+impl Credential {
+    /// Create a new credential.
+    fn new(
+        id: &str,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) -> Credential {
+        Credential {
+            id: id.to_string(),
+            format,
+            ctype,
+            payload,
+        }
+    }
+}
+
+/// Verifiable Digital Credential Collection
+///
+/// This is the main interface to credentials.
+pub struct VdcCollection {
+    storage: Box<dyn StorageManagerInterface>,
+}
+
+impl VdcCollection {
+    /// Create a new credential set.
+    pub fn new(engine: Box<dyn StorageManagerInterface>) -> VdcCollection {
+        VdcCollection { storage: engine }
+    }
+
+    /// Add a credential to the set.
+    pub fn add(
+        &mut self,
+        id: &str,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) {
+        let val;
+
+        match serde_cbor::to_vec(&Credential::new(id, format, ctype, payload)) {
+            Ok(x) => val = x,
+            Err(_) => return,
+        }
+
+        match self.storage.add(Key(id.to_string()), Value(val)) {
+            Ok(()) => {}
+            Err(x) => println!("error adding credential: {x}"),
+        }
```

Comment:
All of these methods should return [`Result`](https://doc.rust-lang.org/std/result/enum.Result.html) instead of ignoring/logging errors.